module/app/upgrades/register.go

```diff
@@ -0,0 +1,29 @@
+package upgrades
+
+import (
+	"github.com/cosmos/cosmos-sdk/types/module"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	distrkeeper "github.com/cosmos/cosmos-sdk/x/distribution/keeper"
+	mintkeeper "github.com/cosmos/cosmos-sdk/x/mint/keeper"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	upgradekeeper "github.com/cosmos/cosmos-sdk/x/upgrade/keeper"
+	bech32ibckeeper "github.com/osmosis-labs/bech32-ibc/x/bech32ibc/keeper"
+
+	v2 "github.com/Gravity-Bridge/Gravity-Bridge/module/app/upgrades/v2"
+)
+
+// RegisterUpgradeHandlers registers handlers for all upgrades
+// Note: This method has crazy parameters because of circular import issues, I didn't want to make a Gravity struct
+// along with a Gravity interface
+func RegisterUpgradeHandlers(
```

Comment:
all these pointer args should probably be nil checked.

Reply:
Checks added